web3/providers/eth_tester/middleware.py

```diff
@@ -218,7 +220,30 @@ def is_named_block(value):
     'eth_syncing': False,
     'eth_mining': False,
     # Net
-    'net_version': '12345',
+    'net_version': '1',
     'net_listening': False,
     'net_peerCount': 0,
 })
+
+
+def default_transaction_fields_middleware(make_request, web3):
+    def middleware(method, params):
+        if method in (
+            'eth_call',
+            'eth_estimateGas',
+            'eth_sendTransaction',
+        ):
+            # import pdb; pdb.set_trace()
+            transaction = params[0]
+            if 'from' not in transaction:
+                if web3.eth.coinbase:
+                    default_from = web3.eth.coinbase
+                elif web3.eth.accounts:
+                    default_from = web3.eth.accounts[0]
+                else:
+                    default_from = None
+                if default_from:
+                    transaction['from'] = web3.eth.coinbase or web3.eth.accounts
```

Comment:
This has the potential to mutate the dictionary from another context.  Can you use `cytoolz.assoc` to remove the mutation?